src/Chainweb/BlockHeaderDB.hs

```diff
@@ -86,37 +83,11 @@ import Chainweb.Version
 import Data.CAS
 import Data.CAS.RocksDB
 
-import Numeric.Additive
-
 -- -------------------------------------------------------------------------- --
 -- Internal
 
 type E = BlockHeader
 
--- -------------------------------------------------------------------------- --
--- Ranked Block Header
-
-newtype RankedBlockHeader = RankedBlockHeader { _getRankedBlockHeader :: BlockHeader }
```

Comment:
The need to separate these types from some of their basic properties, like serializations and textual representations, may hint at a more profound design issue. In my experience this often indicates that something is wrong with the overall design and abstractions in a project. The issue may have existed before and may have only been exposed by this PR.

Type modules are usually needed when data types of large structures (like an AST) are themself are mutually recursive, which is not the case in the Chainweb code base. So, I'd like to better understand why this is needed.

Reply:
There's a huge recursion that I had to cut. The module that deals with genesis blocks is included in several of the BlockHeaderDB / CutDB / WebBlockHeaderDB/etc files, but it wants to validate blocks, which requires a pact service. Now that pact service needs to look up block headers and payloads for rewind+fast-forward on `restore`, it needs a `BlockHeaderDB`.

To cut the cyclical dependency I had to move everything dealing with validation out. I may be able to fold some of them back together but we'll see.

Reply:
Maybe, you can add a special case for the genesis block validation? That one shouldn't require lookup of block headers.

Genesis blocks are validated only once at startup by pact internally.

Reply:
We run into this issue before. At that time we found a simpler solution that just moved a few functions into a submodule instead of moving all types. Not sure if the same is possible here.

Reply:
I think (hope), that it should be possible refactor 'BlockHeaderDb` such that it doesn't depend on pact, without moving that much code and introducing orphans.

If the issue is payload validation, we should provide pact with a version/view of `BlockHeaderDb` that doesn't depend on it. Non-consensus components should should anyways be given a read-only version of BlockHeaderDb. Such a readonly version doesn't need to depend on validation/pact.

Reply:
Yes, that would be a fix for this issue, and we could re-unify the modules. We can clean up later after this changeset is in, the inverse operation of the refactor I did is trivial

Reply:
@gregorycollins @fosskers Let's do this soon. I created an issue here #272.